testSrc/unit/io/flutter/editor/ActiveEditorsOutlineServiceTest.java

```diff
@@ -0,0 +1,214 @@
+    assertThat(service.get(mainPath), equalTo(null));
+    assertThat(listener.editorsChanged, equalTo(0));
+
+    Testing.runOnDispatchThread(() -> getFixture().openFileInEditor(mainFile.getVirtualFile()));
+    final Editor editor = getFixture().getEditor();
+    final EditorTestFixture editorTestFixture = new EditorTestFixture(getProject(), editor, mainFile.getVirtualFile());
+    flutterDas.updateOutline(mainPath, firstFlutterOutline);
+
+    assertThat(listener.outlineChanged.keySet(), hasItem(mainPath));
+    assertThat(listener.outlineChanged.get(mainPath), equalTo(1));
+    assertThat(service.get(mainPath), equalTo(firstFlutterOutline));
+    assertThat(listener.editorsChanged, equalTo(1));
+
+    // Open another file.
+    final PsiFile fileTwo = getFixture().addFileToProject("lib/main_two.dart", fileContents);
+    final String fileTwoPath = fileTwo.getVirtualFile().getCanonicalPath();
+    final EditorTestFixture editorTestFixtureTwo = new EditorTestFixture(getProject(), editor, fileTwo.getVirtualFile());
+    Testing.runOnDispatchThread(() -> getFixture().openFileInEditor(fileTwo.getVirtualFile()));
+    flutterDas.updateOutline(fileTwoPath, secondFlutterOutline);
+
+    assertThat(listener.outlineChanged.keySet(), hasItem(fileTwoPath));
+    assertThat(listener.outlineChanged.get(fileTwoPath), equalTo(1));
+    assertThat(listener.outlineChanged.get(mainPath), equalTo(1));
+    assertThat(service.get(fileTwoPath), equalTo(secondFlutterOutline));
+    // The main file has been closed, so it is no longer there.
+    assertThat(service.get(mainPath), equalTo(null));
+    assertThat(listener.editorsChanged, equalTo(2));
+
+
+    // NOTE: We can't test what happens when closing an editor because CodeInsightTestFixture always has one open editor.
+
+    Testing.runOnDispatchThread(() -> getFixture().openFileInEditor(mainFile.getVirtualFile()));
+    final EditorTestFixture editorTestFixtureThree = new EditorTestFixture(getProject(), editor, mainFile.getVirtualFile());
+    flutterDas.updateOutline(mainPath, firstFlutterOutline);
+
+    assertThat(listener.outlineChanged.get(fileTwoPath), equalTo(1));
+    assertThat(listener.outlineChanged.get(mainPath), equalTo(2));
+    assertThat(service.get(mainPath), equalTo(firstFlutterOutline));
+    // Now file two is closed.
+    assertThat(service.get(fileTwoPath), equalTo(null));
+    assertThat(listener.editorsChanged, equalTo(3));
+  }
+
+  private class Listener implements ActiveEditorsOutlineService.Listener {
+    int editorsChanged = 0;
+    final Map<String, Integer> outlineChanged = new HashMap<>();
+
+    @Override
+    public void onEditorsChanged() {
+      editorsChanged++;
+    }
+
+    @Override
+    public void onOutlineChanged(String path) {
+      final Integer changes = outlineChanged.get(path);
+      outlineChanged.put(path, changes == null ? 1 : changes + 1);
+    }
+  }
+
+  private static class Fixture extends AdaptedFixture<CodeInsightTestFixture> {
```

Comment:
Did you try changing ProjectFixture to inherit from CodeInsightTestFixture instead of IdeaProjectTestFixture? If that causes breakage then I'd like to see this Fixture class moved to a sibling of ProjectFixture, perhaps named CodeInsightProjectFixture (I don't care too much what it is named). It would include getProject() and getModule().

Reply:
Done.